paymentsheet/src/test/java/com/stripe/android/paymentsheet/state/DefaultPaymentSheetLoaderTest.kt

```diff
@@ -603,6 +606,49 @@ internal class DefaultPaymentSheetLoaderTest {
         )
     }
 
+    @Test
+    fun `Disables link sign up if used before`() = runTest {
+        val loader = createPaymentSheetLoader(
+            stripeIntent = PaymentIntentFixtures.PI_REQUIRES_PAYMENT_METHOD,
+            linkSettings = ElementsSession.LinkSettings(
+                linkFundingSources = emptyList(),
+                linkPassthroughModeEnabled = false,
+                linkFlags = mapOf(),
+                disableLinkSignup = false,
+            ),
+            linkStore = mock {
+                on { hasUsedLink() } doReturn true
+            }
+        )
+
+        val result = loader.load(
+            initializationMode = PaymentSheet.InitializationMode.PaymentIntent("secret"),
+            paymentSheetConfiguration = mockConfiguration(),
+        ).getOrThrow()
+
+        assertThat(result.linkState?.configuration?.signupMode).isNull()
+    }
+
+    @Test
+    fun `Disables link sign up when settings have it disabled`() = runTest {
+        val loader = createPaymentSheetLoader(
+            stripeIntent = PaymentIntentFixtures.PI_REQUIRES_PAYMENT_METHOD,
+            linkSettings = ElementsSession.LinkSettings(
+                linkFundingSources = emptyList(),
+                linkPassthroughModeEnabled = false,
+                linkFlags = mapOf(),
+                disableLinkSignup = true,
+            )
+        )
+
+        val result = loader.load(
+            initializationMode = PaymentSheet.InitializationMode.PaymentIntent("secret"),
+            paymentSheetConfiguration = mockConfiguration(),
+        ).getOrThrow()
+
+        assertThat(result.linkState?.configuration?.signupMode).isNull()
```

Comment:
Do we already have a test for it gets enabled in the different states?

Reply:
Yes, we have a few tests already for `signupMode`:
- One for checking `signupMode` is `AlongsideSaveForFutureUse` when `Save for future usage` is enabled.
- One for checking `signupMode` is `InsteadOfSaveForFutureUse` when `Save for future usage` is disabled
- One for ensuring `signupMode` is `null` when Link has already been used.